channeldb/graph.go

```diff
@@ -549,6 +553,11 @@ func (c *ChannelGraph) AddLightningNode(node *LightningNode,
 
 	r := &batch.Request{
 		Update: func(tx kvdb.RwTx) error {
+			wNode := &wrappedNode{node}
+			if err := c.graphCache.AddNode(tx, wNode); err != nil {
+				return err
+			}
+
```

Comment:
I think I've also posed this q on the kv-level cache PR: Isn't desync an issue if the cache is updated here, but the db operation fails?

Reply:
If this is is included in `OnCommit`, then I think we side step this issue, as it's only execute when the transaction executes w/o failure: https://github.com/lightningnetwork/lnd/blob/9e8b9ccd4c72f33e162f9b84b0f4d1f68e855984/batch/scheduler.go#L95

Reply:
The kv level cache must does have exclusive access to the buckets it decorates. We never retry on DB errors but bubble those up to the application level. Retries only happen on conflicts and only inside closed form operations (`View`, `Update`).